app.js

```diff
@@ -1,74 +1,57 @@
-    } else if(reg.active) {
-      console.log('Service worker active');
+const registerServiceWorker = async () => {
+  const registration = await navigator.serviceWorker.register(
+    "/sw-test/sw.js",
+    {
+      scope: "/sw-test/",
     }
+  );
+  if (registration.installing) {
+    console.log("Service worker installing");
+  } else if (registration.waiting) {
+    console.log("Service worker installed");
+  } else if (registration.active) {
+    console.log("Service worker active");
+  }
+};
 
-  }).catch(function(error) {
-    // registration failed
-    console.log('Registration failed with ' + error);
-  });
-}
-
-// function for loading each image via XHR
-
-function imgLoad(imgJSON) {
-  // return a promise for an image loading
-  return new Promise(function(resolve, reject) {
-    var request = new XMLHttpRequest();
-    request.open('GET', imgJSON.url);
-    request.responseType = 'blob';
-
-    request.onload = function() {
-      if (request.status == 200) {
-        var arrayResponse = [];
-        arrayResponse[0] = request.response;
-        arrayResponse[1] = imgJSON;
-        resolve(arrayResponse);
-      } else {
-        reject(Error('Image didn\'t load successfully; error code:' + request.statusText));
-      }
-    };
-
-    request.onerror = function() {
-      reject(Error('There was a network error.'));
-    };
-
-    // Send the request
-    request.send();
-  });
-}
-
-var imgSection = document.querySelector('section');
-
-window.onload = function() {
-
-  // load each set of image, alt text, name and caption
-  for(var i = 0; i<=Gallery.images.length-1; i++) {
-    imgLoad(Gallery.images[i]).then(function(arrayResponse) {
-
-      var myImage = document.createElement('img');
-      var myFigure = document.createElement('figure');
-      var myCaption = document.createElement('caption');
-      var imageURL = window.URL.createObjectURL(arrayResponse[0]);
+const imgSection = document.querySelector("section");
 
-      myImage.src = imageURL;
-      myImage.setAttribute('alt', arrayResponse[1].alt);
-      myCaption.innerHTML = '<strong>' + arrayResponse[1].name + '</strong>: Taken by ' + arrayResponse[1].credit;
+const getImageBlob = async (url) => {
+  const imageResponse = await fetch(url);
+  if (!imageResponse.ok) {
+    throw new Error(
+      "Image didn't load successfully; error code:" + imageResponse.statusText
+    );
+  }
+  return imageResponse.blob();
+};
 
-      imgSection.appendChild(myFigure);
-      myFigure.appendChild(myImage);
-      myFigure.appendChild(myCaption);
+const createGalleryFigure = async (galleryImage) => {
+  const imageBlob = await getImageBlob(galleryImage.url);
+  const myImage = document.createElement("img");
+  const myCaption = document.createElement("caption");
+  const myFigure = document.createElement("figure");
+  myCaption.innerHTML =
+    "<strong>" +
+    galleryImage.name +
+    "</strong>: Taken by " +
+    galleryImage.credit;
```

Comment:
While we are here could we get rid of this `innerHTML`? I think it would be OK to skip the `<strong>` for the name, like:

```suggestion
  myCaption.textContent = `${galleryImage.name}: Taken by ${galleryImage.credit}`;
```

...but if we do want to keep it we could of course create a `<strong>` element, as we do for the other elements here. I'm 50/50 on which to choose here :)

Reply:
I like textContent, done!